Heimdall/Heimdall.swift

```diff
@@ -95,6 +95,19 @@ public class Heimdall {
             completion(result.map { _ in return })
         }
     }
+    
+    /// Requests an access token with the given grant type URI and parameters
+    ///
+    /// **Note:** The completion closure may be invoked on any thread.
+    ///
+    /// :param: grantType The grant type URI of the extension grant
+    /// :param: parameters The required parameters for the external grant
+    /// :param: completion A callback to invoke when the request completed.
+    public func requestAccessToken(#grantType: String, parameters: [String: String], completion: Result<Void, NSError> -> ()) {
```

Comment:
I'd say no. If I read the RFC correctly the extension grant is the only way to implement a custom scheme. The grants we are still missing are "implicit" and "authorization code" and "client credentials". I'd add specific methods for these grants in the future.